lib/tracer/tracer.js

```diff
@@ -276,6 +278,62 @@ Tracer.prototype.buildTrace = function (traceInfo) {
   return traceInfo;
 };
 
+/**
+ * There are two formats for traces. While the method/publication is running, the trace is in the object format.
+ * This is easier to work with, but takes more space to store. After the trace is complete (either finished or errored),
+ * it is built which among other things converts the events to the array format.
+ *
+ * The key difference of `optimizeEvent` and `optimizeEvents` is that they do not mutate the original events.
+ */
+Tracer.prototype.optimizeEvent = function (objectEvent) {
+  const {at, endAt, stack, nested = [], forcedEnd, name} = objectEvent;
+
+  const computeTime = at && endAt ? endAt - at : 0;
```

Comment:
If there isn't an end time, it should use the current time. It should then also set forcedEnd to true to indicate that the event ended early.